test/int/breakOnLoad.test.ts

```diff
@@ -0,0 +1,426 @@
+                const testProjectRoot = path.join(DATA_ROOT, 'breakOnLoad_webpack');
+                const scriptPath = path.join(testProjectRoot, 'src/script.js');
+
+                server = createServer({ root: testProjectRoot });
+                server.listen(7890);
+
+                const url = 'http://localhost:7890/dist/index.html';
+
+                const bpLine = 1;
+                const bpCol = 1;
+
+                await dc.hitBreakpointUnverified({ url, webRoot: testProjectRoot }, { path: scriptPath, line: bpLine, column: bpCol });
+            });
+        });
+    });
+
+    suite('BreakOnLoad Disabled (strategy: none)', () => {
+        let dc: ts.ExtendedDebugClient;
+        setup(() => {
+            return testSetup.setup(undefined, "none")
+                .then(_dc => dc = _dc);
+        });
+
+        let server: any;
+        teardown(() => {
+            if (server) {
+                server.close();
+            }
+
+            return testSetup.teardown();
+        });
+
+        test('Does not hit a breakpoint in a file on load', async () => {
+            const testProjectRoot = path.join(DATA_ROOT, 'breakOnLoad_sourceMaps');
+            const scriptPath = path.join(testProjectRoot, 'src/script.ts');
+
+            server = createServer({ root: testProjectRoot });
+            server.listen(7890);
+
+            const url = 'http://localhost:7890/index.html';
+
+            // We try to put a breakpoint at (1,1). If this doesn't get hit, the console.log statement in the script should be executed
+            const bpLine = 1;
+            const bpCol = 1;
+
+            // Variable to track whether the concerned console.log statement gets executed
+            var consoleOutputLogged = false;
+
+            await Promise.all([
+                dc.waitForEvent('initialized').then(event => {
+                    return dc.setBreakpointsRequest({
+                        lines: [bpLine],
+                        breakpoints: [{ line: bpLine, column: bpCol }],
+                        source: { path: scriptPath }
+                    });
+                }).then(response => {
+                    return dc.configurationDoneRequest();
+                }),
+                // Add an event listener for the output event to capture the console.log statement
+                dc.addListener('output', function(event) {
+                    // If console.log event statement is executed, change the variable to true
+                    if (event.body.category === 'stdout' && event.body.output === 'Hi\n') {
+                        consoleOutputLogged = true;
+                    }
+                }),
+
+                dc.launch({ url: 'http://localhost:7890/index.html', webRoot: testProjectRoot })
+            ]);
+
+            // Wait for sometime to make sure the page load is completed
+            await new Promise(resolve => setTimeout(resolve, 1500));

```

Comment:
You can get rid of the timeout by making the test pass when the console.log message is received.